administrator/components/com_associations/models/fields/itemlanguage.php

```diff
@@ -72,17 +74,40 @@ protected function getOptions()
 				{
 					parse_str($associations[$lang->value], $contents);
 					$removeExtra  = explode(":", $contents['id']);
-					$lang->value  = $lang->value . "|" . $removeExtra[0];
+					$itemId       = $removeExtra[0];
+					$lang->value  = $lang->value . "|" . $itemId;
 				}
 				else
 				{
+					$itemId      = $associations[$lang->value];
 					$lang->value = $lang->value . "|" . $associations[$lang->value];
 				}
+
+				$canEdit    = $user->authorise('core.edit', $component->assetKey . '.' . $itemId);
+				$table->load($itemId);
+
+				if (!is_null($table->{$component->fields->created_by}))
+				{
+					$canEditOwn = $user->authorise('core.edit.own', $component->assetKey . '.' . $itemId) && $table->{$component->fields->created_by} == $user->id;
+					$canEdit    = $canEdit || $canEditOwn;
+				}
+
+				if (!$canEdit)
+				{
+					$lang->disable = true;
+				}	
 			}
 			else
 			{
 				$lang->value .= '|0';
+				$canCreate    = $user->authorise('core.create', $component->assetKey);
+
+				if (!$canCreate)
+				{
+					$lang->disable = true;
+				}
```

Comment:
additional comment for this cycle.

don't you also need to check the checkout state (if componet support)?